### ISSUE TITLE
Add OIDC SSO Support

### ISSUE THREAD

**Issue by tankerkiller125** [sysadminsmedia/homebox#6]
### What is the problem you are trying to solve with this feature?

Authentication with SSO

### What is the solution you are proposing?

Adding OIDC support would immediatly add SSO support from many different authentication providers both open-source and ones like Azure, Okta, etc.

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [ ] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by katosdev** [comment]
Self assigning and I’ll review this and see what I can identify 👍🏻

**Comment by LIRIKKER** [comment]
If this will help get my Authentik instance authing with this then +1 for this.

**Comment by katosdev** [comment]
> If this will help get my Authentik instance authing with this then +1 for this.

It should do, yep. I can’t promise how soon it’ll come though to be honest!

**Comment by NedPool** [comment]
Very much appreciate the work on this! Would love to use my Keycloak with Homebox!

**Comment by katosdev** [comment]
Adding a bounty to get this one over the line.

**Comment by katosdev** [comment]
> boss

Thanks for supporting @LIRIKKER !

**Comment by kloenk** [comment]
Currently looking into this. apparently e.g. Azur has a broken implementation, but I don't have azur and just keycloak to test against

**Comment by kaindlnetwork** [comment]
/boss 10€

I want that feature too :D

**Comment by katosdev** [comment]
I’m very eager to get this in, so I’ll see about getting a review of the current PR done ASAP

**Comment by pnogemane** [comment]
Is there work still needing to be done on the project?

**Comment by kloenk** [comment]
> Is there work still needing to be done on the project?

Yes, still requires proper frontend changes, and a api route to get the oidc urls from the backend and display this in the front end

**Comment by pnogemane** [comment]
> > Is there work still needing to be done on the project?
> 
> Yes, still requires proper frontend changes, and a api route to get the oidc urls from the backend and display this in the front end

Thank you for the update. Should there be any additional feature requirements.  Please feel free to send messages on my dicord server on https://discord.gg/9MG5SERr 

Or email a specification to  devccountpnogemane@gmail.com. I will pull down the repo so long continue with specified changes. Thank you I will keep you updated.

**Comment by rnasci-hub** [comment]
Hi, I have implemented OIDC authentication in both the backend and frontend. I would like to know how you would like me to handle the configuration of the various parameters to be passed to the auth provider. In the backend they are loaded as shown in the figure. I would also like to know if there are any particular rules that I should respect regarding the naming conventions and if I need to make any other graphical changes in addition to adding the new authentication button and editing the profile page. At the moment I have tested it with Google OAuth.![Image](https://github.com/user-attachments/assets/f6412a6b-e9cc-4691-8eb5-fcdbd1f1a868)

**Comment by yankees5963** [comment]
would love to see this implemented

**Comment by kaindlnetwork** [comment]
I increase the motivation by adding 10€ extra as bounty :)

**Comment by Erwan-loot** [comment]
Hello thanks for the feature,
don't forget to add some customizations for the claims of custom providers
For example i'm using synology SSO and there is no claim name but username instead
some project like mealie or homarr added a variable to override the claim 
https://github.com/homarr-labs/homarr/issues/1830
https://github.com/mealie-recipes/mealie/pull/4544/files

**Comment by sunjam** [comment]
Don't see the PR request linked, so:
- https://github.com/sysadminsmedia/homebox/pull/168

**Comment by jimmylemieux** [comment]
I am willing to be a tester if we get this in place. I had to create a workaround with Apache but it's a kluge for sure.

**Comment by famaia** [comment]
Also like to have.

**Comment by 35C4n0r** [comment]
@tankerkiller125 are you actively working on this issue (you're assigned) or is this up for grabs ?

**Comment by tankerkiller125** [comment]
@35C4n0r feel free to go for it, I'm currently focused on some other large areas of the project so I won't have time to get back to OIDC for quite some time. If you want to try your hand and can get it sorted then you'll probably be way ahead of me.

**Comment by kloenk** [comment]
> @35C4n0r feel free to go for it, I'm currently focused on some other large areas of the project so I won't have time to get back to OIDC for quite some time. If you want to try your hand and can get it sorted then you'll probably be way ahead of me.

Did you see that there is also one PR in a somewhat unfinished state but maybe it can be of help

**Comment by UndyingSoul** [comment]
> > @35C4n0r feel free to go for it, I'm currently focused on some other large areas of the project so I won't have time to get back to OIDC for quite some time. If you want to try your hand and can get it sorted then you'll probably be way ahead of me.
> 
> Did you see that there is also one PR in a somewhat unfinished state but maybe it can be of help 

Here's the open PR if you're having a hard time finding it:  https://github.com/sysadminsmedia/homebox/pull/168

Edit: I now see this has been mentioned at least once 😅

**Comment by Adityapradh** [comment]
Hi @tankerkiller125 👋, I'd love to contribute to implementing OIDC-based SSO for this project. I have experience working with OIDC flows and integrating providers like Okta and AzureAD. Would love to discuss initial approach and scope before diving in. Also, how would the bounty/sponsorship process work? Thanks!

**Comment by yannicschuller** [comment]
Working on it. Most likely, I will get it review-ready by tomorrow.

**Comment by yannicschuller** [comment]
Done. Let's wait for the checks to pass and someone from the team to test the PR. If there are any questions, I'm happy to answer them.

**Comment by AJL1-Hub** [comment]
Has their been any more progress made on this?

**Comment by tankerkiller125** [comment]
There are actually two implementations currently in PR, #996 and #1023

**Comment by tankerkiller125** [comment]
@slipperybeluga @katosdev @LIRIKKER @kaindlnetwork @sushain97 @IlyaVassyutovich @revellion @jmiller4038 @KF5JWC @greirson @andrewgazelka

In case you didn't see it [our bounty program is currently very broken](https://github.com/sysadminsmedia/homebox/discussions/1033), and as such when this issue gets closed by a PR the bounty wouldn't ever get paid out.

We're still working on a long term permanent solution for bounties overall, however, given this issue already has several PR attempts by various authors we want to make sure that if you want to give a bounty for this issue that you have that chance and that it will get paid out.

As a reminder, any bounty set in bounty.dev has not been paid, and as it currently stands, will never get paid (both because boss.dev is broken, and because we disabled the integration).

You can contribute to the bounty at https://opencollective.com/homebox/projects/issue-6

All funds in the project will be given to the contributor of the PR that gets merged.

**Comment by revellion** [comment]
Ah thanks for the heads-up, i'll put in a donation into the opencollective link instead 💃

**Comment by KF5JWC** [comment]
Thank you. Contribution moved 👍

**Comment by verstehbahnhof** [comment]
Have also contributed via opencollective. We are really really craving this feature!

Thanks generally for picking up development on this amazing project.

daniel
